infra/profile.sh

```diff
@@ -11,37 +11,48 @@ trap cleanup EXIT
 # TODO: take in file glob as command line argument
 PROFILES=tests/small/add.bril
 
+RUNMODES=("nothing" "rvsdg-optimize")
+
 # create temporary directory structure necessary for bench runs
 mkdir -p ./tmp/bench
-mkdir ./tmp/hyperfine
 
 # bench will benchmark a single bril file, outputting its contents to ./tmp/bench/<profile_name>.json
 bench() {
-    profile=$1
+    profile="../../../$1"
 
     # strip the file path down to just the file name
     # TODO: profile name is not unique, generate a unique output path (it will be aggregated anyway)
-    profile_name=$(basename -- $profile)
+    profile_file=$(basename -- $profile)
+    profile_name="${profile_file%.*}"
+
+    mkdir ./tmp/bench/"$profile_name"
+    pushd ./tmp/bench/"$profile_name"
 
-    # run eggcc, interp the program and put the data out to $out
-    out="./tmp/bench/${profile_name}.json"
+		# loop over RUNMODES and generate a profile for each, leaving it in the profile_name directory
+    for mode in ${RUNMODES[@]}
+		do
+		  echo "profiling $mode"
 
-    cargo run --release $profile --interp --profile-out "$out"
+			out="${mode}.json"
+    	pwd
+			# generate the instruction count profile by interp'ing
+			cargo run --release $profile --interp --profile-out="./$out" --run-mode "$mode"
+			
+	    # $out now contains a key value of total_dyn_inst: value, so use read to get the key/value
+      # TODO: this is kind of a yaml sort of format so maybe yq would be good in the future
+			IFS=': ' read KEY VAL <<< $(cat $out)
 
-    # $out now contains a key value of total_dyn_inst: value, so use read to get the key/value
-    # TODO: this is kind of a yaml sort of format so maybe yq would be good in the future
-    read KEY VAL <<< $(cat $out)
+			# generate the profile and overrite the $out file
+			hyperfine --warmup 2 --export-json "$out" "cargo run --release $profile --interp --run-mode $mode"
 
-    # export hyperfine out to tmp file
-    hyperfine_out="./tmp/hyperfine/${profile_name}.json"
-    hyperfine --warmup 2 --export-json "$hyperfine_out" "cargo run --release $profile --interp"
+			# overwwrite outfile with json version of profile data, annotate with profile name.
+    	# we also combine both instruction count and hyperfine json output into a single object
+    	# to make things super easy
+			printf '{"%s": "%s", "hyperfine": %s}' \
```

Comment:
would prefer you generate a file with a different name instead of overwriting

Reply:
If I'm gonna do that `read` in Python, then yes, I'll want a different file